fw/http_frame.c

```diff
@@ -964,17 +964,17 @@ tfw_h2_headers_process(TfwH2Ctx *ctx)
 
 		ctx->state = HTTP2_IGNORE_FRAME_DATA;
 
-		if (!STREAM_SEND_PROCESS(ctx->cur_stream, HTTP2_RST_STREAM, 0))
-			return tfw_h2_stream_close(ctx, hdr->stream_id,
-						   &ctx->cur_stream,
-						   HTTP2_ECODE_PROTO);
-		return T_OK;
+		if (STREAM_SEND_PROCESS(ctx->cur_stream, HTTP2_RST_STREAM, 0))
+			return T_BAD;
```

Comment:
Please pay attention. May be you have idea about more appropriate error code here from errno.h